_posts/2021-11-11-debezium-1.8-alpha2-released.adoc

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title:  Debezium 1.8.0.Alpha2 Released
+date:   2021-11-11
+tags: [ releases, mysql, postgres, sqlserver, cassandra, oracle, db2, vitess, outbox ]
+author: ccranfor
+---
+
+It's my pleasure to announce the second release of the Debezium 1.8 series, *1.8.0.Alpha2*!
+
+With the holiday season just around the corner, the team's release schedule remains steadfast.
+While Debezium 1.8.0.Alpha2 delivers quite a lot of bugfixes and minor changes, there are a few notable changes:
+
+* MySQL support for heartbeat action queries
+* Configurable transaction topic name
+
++++<!-- more -->+++
+
+[NOTE]
+====
+This release does include several breaking changes.
+Please see the https://debezium.io/releases/1.8/release-notes#release-1.8.0-alpha2[release notes] for details on what changed and how to upgrade.
+====
+
+== MySQL heartbeat action query support
+
+A heartbeat action query can be enabled by supplying a `heartbeat.action.query` configuration option in the connector's configuration.
+This property is meant to supply a SQL statement that the connector will execute periodically.
+
+The initial implementation of the heartbeat action query was specifically for PostgreSQL to handle dealing with WAL growth under specific conditions.
+But the use of a heartbeat action query has many uses and is entirely connector or even user specific.
+
+For MySQL, you may want to notify downstream consumers that your MySQL topology has changed by supplying consumers with an event with the GTID.
+The following configuration illustrates capturing changes from the heartbeat action query table that can then be consumed easily by your CDC pipeline.
+
+[source,yaml]
+----
+"table.include.list": "gtid_history",
+"heartbeat.action.query": "INSERT INTO gtid_history( select * from mysql.gtid_executed )"
+----
+
+== Configurable transaction topic names
+
+The Debezium transaction metadata topic had previously used a relatively non-configurable naming convention of `<database.server.name>.transaction`.
+While it was possible to manipulate the topic name using a single message transform (SMT) as a workaround, we felt that allowing this to be a bit more flexible in Debezium proper made sense.
+
+A new configuration option, `transaction.topic.prefix`, has been introduced that allows the connector configuration to adjust the naming of the transaction metadata topic.
+The configuration option value specifies what will be used as a direct replacement for the `<database.server.name>~ portion of the topic name.
+If this configuration option is not supplied, the prior topic naming behavior will continue to be used; requiring no changes for existing connector deployments.
+
+== Other Fixes
+
+There were quite a number of bugfixes and stability changes in this release, some noteworthy are:
+
+* Invalid default value error on captured table DDL with default value [DBZ-3710](https://issues.jboss.org/browse/DBZ-3710)
```

Comment:
Links are in Markdown syntax rather than AsciiDoc. Happens to me all the time too :) While fixing, can you also point to the redhat.com URL instead of jboss.org.

Reply:
Ah my apologies, I took these directly from the release notes adoc on the website, but I'll adjust the announcement.

Reply:
Mh, interesting. Just seeing these links are broken there over too. Could you do the same adjustment (AsciiDoc syntax, redhat.com link) for the release notes?